src/core/mac/mac_frame.cpp

```diff
@@ -1232,19 +1232,34 @@ const uint8_t *Frame::GetThreadIe(uint8_t aSubType) const
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
 void Frame::SetCslIe(uint16_t aCslPeriod, uint16_t aCslPhase)
 {
-    uint8_t *cur = GetHeaderIe(CslIe::kHeaderIeId);
-    CslIe   *csl;
+    CslIe *csl = GetCslIe();
 
-    VerifyOrExit(cur != nullptr);
-
-    csl = reinterpret_cast<CslIe *>(cur + sizeof(HeaderIe));
+    VerifyOrExit(csl != nullptr);
     csl->SetPeriod(aCslPeriod);
     csl->SetPhase(aCslPhase);
+
 exit:
     return;
 }
+
+bool Frame::IsCslIePresent(void) const { return GetHeaderIe(CslIe::kHeaderIeId) != nullptr; }
 #endif // OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
 
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE || OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+const CslIe *Frame::GetCslIe(void) const
+{
+    const uint8_t *cur;
+    const CslIe   *csl = nullptr;
+
+    cur = GetHeaderIe(CslIe::kHeaderIeId);
+    VerifyOrExit(cur != nullptr);
+    csl = reinterpret_cast<const CslIe *>(cur + sizeof(HeaderIe));
+
+exit:
+    return csl;
+}
+#endif // OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE || OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```

Comment:
```suggestion
#endif
```
Since it's < 20 lines we can skip the comment here (it will be long after updating as above mentioned)